lib/SIL/Verifier/SILVerifier.cpp

```diff
@@ -7192,6 +7192,8 @@ void SILProperty::verify(const SILModule &M) const {
   auto sig = dc->getGenericSignatureOfContext();
   auto baseTy = dc->getInnermostTypeContext()->getSelfInterfaceType()
                   ->getReducedType(sig);
+  if (decl->isStatic())
+    baseTy = CanMetatypeType::get(baseTy);
```

Comment:
Since retrieval of the base type is not as straightforward anymore I think it should either become a property of `SILProperty` or we should add a method on SILProperty i.e. `CanType SILProperty::getBaseType() const` to compute it internally.

Reply:
I added `CanType SILProperty::getBaseType() const` to SILVerifier.cpp above ::verify and don't think that is the place for it but am not sure where else it should go. Perhaps at the end of SILType.cpp?